### PR TITLE
update mkimage-yum.sh to support dnf

### PR DESCRIPTION
**Pull request by ekuric**
Updated mkimage-yum.sh to to support dnf - for Fedora users, currently mkimage-yum.sh only supports yum. Fedora 22 and never use dnf

### CODE REVIEW

File: contrib/mkimage-yum.sh [hunk 1]
```diff
@@ -11,13 +11,18 @@ usage() {
...
+                default is /etc/yum.conf for Centos/RHEL and /etc/dnf/dnf.conf for Fedora 
 EOOPTS
     exit 1
 }
 
 # option defaults
-yum_config=/etc/yum.conf
+
+if [ -f /etc/dnf/dnf.conf ] ; then
+	yum_config=/etc/dnf/dnf.conf
```
Comment (tianon) [inline contrib/mkimage-yum.sh]: Couldn't we just add `alias yum=dnf` here and skip the `if` down below?

**Comment by ekuric** [comment]
@tianon hm... not sure.But, how do you mean, where in mkimage-yum.sh to do this? I think yum and dnf does not follow same /etc structure so we need if/else, but if there is better approach, please apply it.

**Comment by thaJeztah** [comment]
@ekuric could you also make sure that your commit(s) are signed of properly?

**Comment by ekuric** [comment]
@thaJeztah yes, I forgot that last time will respect that rule in future.Thank you for reminder.

**Comment by thaJeztah** [comment]
ping @tianon could you explain how you wanted to do https://github.com/docker/docker/pull/17362#discussion_r43017450?

(thanks for fixing the sign-off @ekuric )

**Comment by LK4D4** [comment]
ping @tianon

File: contrib/mkimage-yum.sh [hunk 2]
```diff
@@ -63,9 +68,15 @@ if [ -d /etc/yum/vars ]; then
...
-    --setopt=group_package_types=mandatory -y groupinstall Core
-yum -c "$yum_config" --installroot="$target" -y clean all
+if [ -f /etc/dnf/dnf.conf ] ; then 
+	dnf -c "$yum_config" --installroot="$target" --releasever=/ --setopt=tsflags=nodocs \
+		--setopt=group_package_types=mandatory -y groupinstall Core
+	dnf -c "$yum_config" --installroot="$target" -y clean all
+else
+	yum -c "$yum_config" --installroot="$target" --releasever=/ --setopt=tsflags=nodocs \
+    		--setopt=group_package_types=mandatory -y groupinstall Core
+	yum -c "$yum_config" --installroot="$target" -y clean all
```
Comment (tianon) [inline contrib/mkimage-yum.sh]: What I meant with my previous comment was that the two code blocks in this `if` statement are _identical_ sans the change from `yum` to `dnf`, so I'd much rather see us do this check up higher and either have a variable of `yum=dnf` and use `$yum` here, or just do `alias yum=dnf` above so that `yum` isn't invoked.

ie:
```bash
yum_config=/etc/yum.conf
if [ -f /etc/dnf/dnf.conf ] && command -v dnf &> /dev/null; then
	yum_config=/etc/dnf/dnf.conf
	alias yum=dnf
fi

# ...

yum -c "$yum_config" --installroot="$target" --releasever=/ --setopt=tsflags=nodocs \
	--setopt=group_package_types=mandatory -y groupinstall Core
yum -c "$yum_config" --installroot="$target" -y clean all
```

**Comment by ekuric** [comment]
@tianon @thaJeztah I changed file per last @tianon  proposal / advice. Thank you

File: contrib/mkimage-yum.sh [hunk 3]
```diff
@@ -64,7 +70,7 @@ if [ -d /etc/yum/vars ]; then
 fi
 
 yum -c "$yum_config" --installroot="$target" --releasever=/ --setopt=tsflags=nodocs \
-    --setopt=group_package_types=mandatory -y groupinstall Core
+    		--setopt=group_package_types=mandatory -y groupinstall Core
```
Comment (thaJeztah) [inline contrib/mkimage-yum.sh]: nit: unrelated change?

**Comment by thaJeztah** [comment]
@ekuric could you squash your commits so that there's only a single commit in this PR, and make sure the commit is signed-off? If you need help with that; there's a section in our contribution guide explaining how to; http://docs.docker.com/opensource/workflow/work-issue/#pull-and-rebase-frequently

**Comment by LK4D4** [comment]
ping @ekuric

**Comment by ekuric** [comment]
@LK4D4 @thaJeztah only 4288fc2 is valid, I tried to clean up this git stuff, but there are stil 73195f0, e7b9a65, e07a78c which I do not want there, but not sure which git magic will clean up them. 
commit 4288fc2 has all what @tianon proposed regarding mkimage-yum.sh, and we just need it,  ... sorry and thank you

**Comment by thaJeztah** [comment]
@ekuric feel free to ask for assistance in the #docker-dev IRC channel. I'm not there at the moment, but I'm sure someone is there to help out

**Comment by LK4D4** [comment]
@ekuric You can use `git rebase -i` to remove unwanted commit. https://help.github.com/articles/about-git-rebase/
You need to rebase against commit in same branch(last before your changes).

**Comment by ekuric** [comment]
@thaJeztah I guess easier to close this one,all work we had here is in #18142.Thank you!

**Comment by thaJeztah** [comment]
Thanks @ekuric !
